### PR TITLE
[Feature] Add environment variable to switch on/off MXNet set_np_shape

### PR DESCRIPTION
**Pull request by hetong007**
## Description
Fix for https://github.com/dmlc/dgl/issues/1186

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR

### CODE REVIEW

**Comment by hetong007** [comment]
May I treat the errors as flaky tests? @jermainewang

File: python/dgl/backend/mxnet/tensor.py [hunk 1]
```diff
@@ -17,7 +18,10 @@
 
 # After MXNet 1.5, empty tensors aren't supprted by default.
 # After we turn on the numpy compatible flag, MXNet supports empty NDArray.
-mx.set_np_shape(True)
+dgl_mx_np_shape = os.environ.get('DGL_MXNET_NP_SHAPE', '1')
+if dgl_mx_np_shape.isdigit():
+    if int(dgl_mx_np_shape) == 1:
+        mx.set_np_shape(True)
```
Comment (jermainewang) [inline python/dgl/backend/mxnet/tensor.py]: I think it's better just throw error if the env var is not of the required type. How about just:
```python
mx.set_np_shape(bool(os.environ.get('DGL_MXNET_SET_NP_SHAPE', True)))
```

Comment (hetong007) [reply python/dgl/backend/mxnet/tensor.py]: Sounds easier

**Comment by jermainewang** [comment]
One more thing. Would you mind also updating the document [here](https://docs.dgl.ai/install/backend.html) about how to use this env var?

**Comment by jermainewang** [comment]
@hetong007 The PR LGTM. However, it seems that I cannot update this PR with the lastest changes in master branch so I cannot merge it.

**Comment by hetong007** [comment]
Just updated, also ticked "Allow edits from maintainers".

**Pull request merged**